lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1963,5 +1963,61 @@ sub add_to_temporary_input_files_queue {
     return 1;
 }
 
+sub lock_bam_file_access {
+    my $self = shift;
+
+    unless ($self->get_merged_alignment_results) {
+        my @bams = $self->alignment_bam_file_paths;
+        unless (@bams) {
+            die $self->error_message("Alignment result with class (%s) and id (%s) has neither ".
```

Comment:
Unless we have already locked, could the merged alignment result remove the BAM between `$self->get_merged_alignment_results` and `$self->alignment_bam_file_paths` and trigger this `die`?